### PR TITLE
refactor #1497: migrated twofactor package to kotlin

### LINKED ISSUE #1496: refactor: migrate savedcards of core module to kotlin
migrate savedcards usecase within usecase package of core module

### LINKED ISSUE #1497: refactor: migrate twofactor package to kotlin
migrate twofactor usecase within usecase package of core module

### PR DESCRIPTION
**Pull request by PratyushSingh07**
## Issue Fix
Fixes #1497 
Fixes #1496 

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/AddCard.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * Created by ankur on 21/May/2018
+ */
+class AddCard @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<AddCard.RequestValues?, AddCard.ResponseValue?>() {
+    class RequestValues(val clientId: Long, val card: Card) : UseCase.RequestValues
+    class ResponseValue : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/AddCard.kt]: Remove nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/AddCard.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import rx.Subscriber
+import rx.android.schedulers.AndroidSchedulers
+import rx.schedulers.Schedulers
+import javax.inject.Inject
+
+/**
+ * Created by ankur on 21/May/2018
+ */
+class AddCard @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<AddCard.RequestValues?, AddCard.ResponseValue?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/AddCard.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/DeleteCard.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import rx.Subscriber
+import rx.android.schedulers.AndroidSchedulers
+import rx.schedulers.Schedulers
+import javax.inject.Inject
+
+/**
+ * Created by ankur on 22/May/2018
+ */
+class DeleteCard @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<DeleteCard.RequestValues?, DeleteCard.ResponseValue?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/DeleteCard.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/DeleteCard.kt [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    class RequestValues(val clientId: Int, val cardId: Int) : UseCase.RequestValues
+    class ResponseValue : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/DeleteCard.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/EditCard.kt [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import rx.Subscriber
+import rx.android.schedulers.AndroidSchedulers
+import rx.schedulers.Schedulers
+import javax.inject.Inject
+
+/**
+ * Created by ankur on 22/May/2018
+ */
+class EditCard @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<EditCard.RequestValues?, EditCard.ResponseValue?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/EditCard.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/EditCard.kt [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+    class RequestValues(val clientId: Int, val card: Card) : UseCase.RequestValues
+    class ResponseValue : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/EditCard.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/FetchSavedCards.kt [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+import rx.Subscriber
+import rx.android.schedulers.AndroidSchedulers
+import rx.schedulers.Schedulers
+import javax.inject.Inject
+
+/**
+ * Created by ankur on 21/May/2018
+ */
+class FetchSavedCards @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<FetchSavedCards.RequestValues?, FetchSavedCards.ResponseValue?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/FetchSavedCards.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/FetchSavedCards.kt [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+    class RequestValues(val clientId: Long) : UseCase.RequestValues
+    class ResponseValue(val cardList: List<Card?>) : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues?) {
+        if (requestValues != null) {
+            mFineractRepository.fetchSavedCards(requestValues.clientId)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(object : Subscriber<List<Card?>?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/savedcards/FetchSavedCards.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/FetchDeliveryMethods.kt [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import rx.Subscriber
+import rx.android.schedulers.AndroidSchedulers
+import rx.schedulers.Schedulers
+import javax.inject.Inject
+
+/**
+ * Created by ankur on 01/June/2018
+ */
+class FetchDeliveryMethods @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<FetchDeliveryMethods.RequestValues?, FetchDeliveryMethods.ResponseValue?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/FetchDeliveryMethods.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/FetchDeliveryMethods.kt [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    class RequestValues : UseCase.RequestValues
+    class ResponseValue(
+        val deliveryMethodList: List<DeliveryMethod?>
+    ) : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues?) {
+        mFineractRepository.deliveryMethods
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribeOn(Schedulers.io())
+            .subscribe(object : Subscriber<List<DeliveryMethod?>?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/FetchDeliveryMethods.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/FetchDeliveryMethods.kt [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+ */
+class FetchDeliveryMethods @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<FetchDeliveryMethods.RequestValues?, FetchDeliveryMethods.ResponseValue?>() {
+
+    class RequestValues : UseCase.RequestValues
+    class ResponseValue(
+        val deliveryMethodList: List<DeliveryMethod?>
+    ) : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/FetchDeliveryMethods.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/RequestOTP.kt [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+import rx.Subscriber
+import rx.android.schedulers.AndroidSchedulers
+import rx.schedulers.Schedulers
+import javax.inject.Inject
+
+/**
+ * Created by ankur on 01/June/2018
+ */
+class RequestOTP @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<RequestOTP.RequestValues?, RequestOTP.ResponseValue?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/RequestOTP.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/RequestOTP.kt [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+
+    class RequestValues(val deliveryMethod: String) : UseCase.RequestValues
+    class ResponseValue(val response: String) : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/RequestOTP.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/RequestOTP.kt [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+        requestValues?.let {
+            mFineractRepository.requestOTP(it.deliveryMethod)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(object : Subscriber<String?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/RequestOTP.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/ValidateOTP.kt [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+import rx.Subscriber
+import rx.android.schedulers.AndroidSchedulers
+import rx.schedulers.Schedulers
+import javax.inject.Inject
+
+/**
+ * Created by ankur on 01/June/2018
+ */
+class ValidateOTP @Inject constructor(private val mFineractRepository: FineractRepository) :
+    UseCase<ValidateOTP.RequestValues?, ValidateOTP.ResponseValue?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/ValidateOTP.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/ValidateOTP.kt [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+
+    class RequestValues(val token: String) : UseCase.RequestValues
+    class ResponseValue(val accessToken: AccessToken) : UseCase.ResponseValue
+
+    override fun executeUseCase(requestValues: RequestValues?) {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/ValidateOTP.kt]: Remove this nullable `?`

File: core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/ValidateOTP.kt [hunk 17]
```diff
@@ -0,0 +1,37 @@
...
+        if (requestValues != null) {
+            mFineractRepository.validateToken(requestValues.token)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(object : Subscriber<AccessToken?>() {
```
Comment (therajanmaurya) [inline core/data/src/main/java/org/mifos/mobilewallet/core/domain/usecase/twofactor/ValidateOTP.kt]: Remove this nullable `?`

**Review by therajanmaurya — CHANGES_REQUESTED** [review]

**Comment by PratyushSingh07** [comment]
done the requested changes @therajanmaurya

**Pull request merged**